plugins/woocommerce-admin/client/marketplace/components/product-card/product-price.tsx

```diff
@@ -0,0 +1,58 @@
+/**
+ * External dependencies
+ */
+import { Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useContext } from '@wordpress/element';
+import { navigateTo, getNewPath } from '@woocommerce/navigation';
+
+/**
+ * Internal dependencies
+ */
+import { InstallFlowContext } from '~/marketplace/contexts/install-flow-context';
+import { Product } from '../product-list/types';
+
+export default function ProductPrice( props: { product: Product } ) {
+	const { product } = props;
+
+	// We hardcode this for now while we only display prices in USD.
+	const currencySymbol = '$';
+
+	const { setProduct } = useContext( InstallFlowContext );
+
+	function openInstallModal() {
+		navigateTo( {
+			url: getNewPath( { install: 'new' } ),
+		} );
+
+		setProduct( props.product );
```

Comment:
`setProduct` function puts the product object into the context, where I can read it from the connect modal. I use that to show product icon and name in the connect modal. 

However, during the connect process, the page is reloaded and we lose all of our state when we get back. We need to put product name, icon and ID to the URL bar, so even if we get back we know which product to install. I'll change this.